test/helpers.py

```diff
@@ -136,61 +159,270 @@ def _batch_sde_solve(
+            dt,
+            step_controller,
+            bm_tol,
+            saveat,
+        )
+        errs = path_l2_dist(sols, correct_sols)
+        errs_list.append(errs)
+        steps_list.append(jnp.average(steps))
+    errs_arr = jnp.array(errs_list)
+    steps_arr = jnp.array(steps_list)
+    order, _ = jnp.polyfit(jnp.log(1 / steps_arr), jnp.log(errs_arr), 1)
+    return steps_arr, errs_arr, order
+
+
+# TODO: remove this once we have a better way to handle this
+# I understand you'd prefer not to have this in the library and
+# I agree this is somewhat hacky, but I think passing each of
+# these args around separately is a bit of a pain. If this only appeared
+# in the tests, I'd be fine with it, but it's also in the examples
+# (e.g. srk_example.py) and I'd prefer if it looked a bit cleaner there.
+# So how do you recommend we streamline this?
+@dataclasses.dataclass
+class SDE:
+    get_terms: Callable[[AbstractBrownianPath], AbstractTerm]
+    args: PyTree
+    y0: PyTree[Array]
+    t0: float
+    t1: float
+    w_shape: tuple[int, ...]
+
+    def get_dtype(self):
+        return jnp.result_type(*jtu.tree_leaves(self.y0))
+
+    def get_bm(
+        self,
+        bm_key,
+        levy_area: type[diffrax.BrownianIncrement],
+        tol=2**-14,
+    ):
+        shp_dtype = jax.ShapeDtypeStruct(self.w_shape, dtype=self.get_dtype())
+        return VirtualBrownianTree(self.t0, self.t1, tol, shp_dtype, bm_key, levy_area)
+
+
+# A more concise function for use in the examples
+def simple_sde_order(
+    keys,
+    sde: SDE,
+    solver,
+    ref_solver,
+    levels,
+    get_dt_step_controller,
+    saveat,
+    bm_tol,
+):
+    return sde_solver_strong_order(
+        keys,
+        sde.get_terms,
+        sde.w_shape,
+        sde.t0,
+        sde.t1,
+        sde.y0,
+        sde.args,
+        solver,
+        ref_solver,
+        levels,
+        levels[1] + 2,
+        get_dt_step_controller,
+        saveat,
+        bm_tol,
+    )
+
+
+def simple_batch_sde_solve(
+    keys, sde: SDE, solver, levy_area, dt0, controller, bm_tol, saveat
+):
+    return _batch_sde_solve(
+        keys,
+        sde.get_terms,
+        sde.w_shape,
+        sde.t0,
+        sde.t1,
+        sde.y0,
+        sde.args,
+        solver,
+        levy_area,
+        dt0,
+        controller,
+        bm_tol,
+        saveat,
+    )
+
+
+def _squareplus(x):
+    return 0.5 * (x + jnp.sqrt(x**2 + 4))
+
+
+def drift(t, y, args):
+    mlp, _, _ = args
+    return 0.25 * mlp(y)
+
+
+def diffusion(t, y, args):
+    _, mlp, noise_dim = args
+    return 1.0 * mlp(y).reshape(3, noise_dim)
+
+
+def get_mlp_sde(t0, t1, dtype, key, noise_dim):
+    driftkey, diffusionkey, ykey = jr.split(key, 3)
+    # To Patrick: I had to increase the depth of these MLPs, otherwise many SDE
+    # solvers had order ~0.72 which is more than 0.5 + 0.2, which is the maximal
+    # tolerated order. I think the issue was that it was too linear and too easy.
```

Comment:
Makes sense to me!
(Rather than putting these comments in the code by the way, better to leave them as review comments on your own pull request! Then you won't need to remove them afterwards.)